plugins/inputs/influxdb_listener/influxdb_listener.go

```diff
@@ -0,0 +1,404 @@
+	return "InfluxDB listener"
+}
+
+func (h *InfluxDBListener) Gather(_ telegraf.Accumulator) error {
+	return nil
+}
+
+func (h *InfluxDBListener) routes() {
+	authHandler := internal.AuthHandler(h.BasicUsername, h.BasicPassword, "influxdb",
+		func(_ http.ResponseWriter) {
+			h.authFailures.Incr(1)
+		},
+	)
+
+	h.mux.Handle("/write", authHandler(h.handleWrite()))
+	h.mux.Handle("/query", authHandler(h.handleQuery()))
+	h.mux.Handle("/ping", h.handlePing())
+	h.mux.Handle("/", authHandler(h.handleDefault()))
+}
+
+func (h *InfluxDBListener) Init() error {
+	tags := map[string]string{
+		"address": h.ServiceAddress,
+	}
+	h.bytesRecv = selfstat.Register("influxdb_listener", "bytes_received", tags)
+	h.requestsServed = selfstat.Register("influxdb_listener", "requests_served", tags)
+	h.writesServed = selfstat.Register("influxdb_listener", "writes_served", tags)
+	h.queriesServed = selfstat.Register("influxdb_listener", "queries_served", tags)
+	h.pingsServed = selfstat.Register("influxdb_listener", "pings_served", tags)
+	h.requestsRecv = selfstat.Register("influxdb_listener", "requests_received", tags)
+	h.notFoundsServed = selfstat.Register("influxdb_listener", "not_founds_served", tags)
+	h.buffersCreated = selfstat.Register("influxdb_listener", "buffers_created", tags)
+	h.authFailures = selfstat.Register("influxdb_listener", "auth_failures", tags)
+	h.routes()
+
+	if h.MaxBodySize.Size == 0 {
+		h.MaxBodySize.Size = defaultMaxBodySize
+	}
+
+	if h.ReadTimeout.Duration < time.Second {
+		h.ReadTimeout.Duration = time.Second * 10
+	}
+	if h.WriteTimeout.Duration < time.Second {
+		h.WriteTimeout.Duration = time.Second * 10
+	}
+
+	return nil
+}
+
+// Start starts the InfluxDB listener service.
+func (h *InfluxDBListener) Start(acc telegraf.Accumulator) error {
+	h.acc = acc
+
+	tlsConf, err := h.ServerConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	h.server = http.Server{
+		Addr:         h.ServiceAddress,
+		Handler:      h,
+		ReadTimeout:  h.ReadTimeout.Duration,
+		WriteTimeout: h.WriteTimeout.Duration,
+		TLSConfig:    tlsConf,
+	}
+
+	var listener net.Listener
+	if tlsConf != nil {
+		listener, err = tls.Listen("tcp", h.ServiceAddress, tlsConf)
+		if err != nil {
+			return err
+		}
+	} else {
+		listener, err = net.Listen("tcp", h.ServiceAddress)
+		if err != nil {
+			return err
+		}
+	}
+	h.listener = listener
+	h.port = listener.Addr().(*net.TCPAddr).Port
+
+	go func() {
+		err = h.server.Serve(h.listener)
+		if err != http.ErrServerClosed {
+			h.log.Infof("Error serving HTTP on %s", h.ServiceAddress)
+		}
+	}()
+
+	h.log.Infof("Started HTTP listener service on %s", h.ServiceAddress)
+
+	return nil
+}
+
+// Stop cleans up all resources
+func (h *InfluxDBListener) Stop() {
+	err := h.server.Shutdown(context.Background())
+	if err != nil {
+		h.log.Infof("Error shutting down HTTP server: %v", err.Error())
+	}
+
+	h.log.Infof("Stopped HTTP listener service on %s", h.ServiceAddress)
```

Comment:
Let's remove this log message.  Usually we don't have plugins announce they are starting/stopping with the exception of plugins that listen on a port, but for these we usually only announce the listening.

Reply:
done